exchanges/binanceus/binanceus_test.go

```diff
@@ -1756,3 +1760,50 @@ func TestGetCurrencyTradeURL(t *testing.T) {
 		assert.NotEmpty(t, resp)
 	}
 }
+
+// TestGetAggregatedTradesBatched exercises TestGetAggregatedTradesBatched to ensure our date and limit scanning works correctly
+// This test is susceptible to failure if volumes change a lot, during wash trading or zero-fee periods
+// In live tests, 45 minutes is expected to return more than 1000 records
+func TestGetAggregatedTradesBatched(t *testing.T) {
+	t.Parallel()
+	type testCase struct {
+		name    string
+		args    *AggregatedTradeRequestParams
+		expFunc func(*testing.T, []AggregatedTrade)
+	}
+
+	var tests []testCase
+	start := time.Now().Add(-time.Hour * 24 * 90).Truncate(time.Minute) // 3 months ago
+	tests = []testCase{
+		{
+			name: "batch with timerange",
+			args: &AggregatedTradeRequestParams{StartTime: start, EndTime: start.Add(20 * time.Minute)},
+			expFunc: func(t *testing.T, results []AggregatedTrade) {
+				t.Helper()
+				require.NotEmpty(t, results, "must have records")
+				assert.Less(t, len(results), 50000, "should return a quantity below a sane threshold of records")
+				assert.WithinDuration(t, results[len(results)-1].TimeStamp.Time(), start, 20*time.Minute, "last record should be within range of start time")
+			},
+		},
+		{
+			name: "custom limit with start time set, no end time",
+			args: &AggregatedTradeRequestParams{StartTime: start, Limit: 2042},
+			expFunc: func(t *testing.T, results []AggregatedTrade) {
+				t.Helper()
+				// 2000 records in was about 6 minutes in 2025; Adjust if Binance enters a phase of zero-fees or low-volume
+				require.Equal(t, 2042, len(results), "must return exactly the limit number of records")
+				assert.WithinDuration(t, results[len(results)-1].TimeStamp.Time(), start, 48*time.Hour, "last record should be within 20 minutes of start time")
```

Comment:
```suggestion
				assert.WithinDuration(t, results[len(results)-1].TimeStamp.Time(), start, 48*time.Hour, "last record should be within 48 hours of start time")
```
binanceus is the place for trading solo

Reply:
Reviewed all of the numbers with live results and updated accordingly.

Fixed https://github.com/gbjk/gocryptotrader/commit/7fd911d1b

If this turns out to be fragile can just widen the hole ....